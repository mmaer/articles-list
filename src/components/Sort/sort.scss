.sort {
  text-align: right;

  &__arrows {
    padding-right: 18px;
    position: relative;
    width: 100%;
    color: #000;

    &:before,
    &:after {
      border: 6px solid transparent;
      content: '';
      display: block;
      height: 0;
      right: 5px;
      top: 50%;
      position: absolute;
      width: 0;
    }

    &:before {
      margin-top: -12px;
      border-bottom-color: #666;
    }

    &:after {
      margin-top: 2px;
      border-top-color: #666;
    }

    &:hover:before {
      cursor: pointer;
      border-bottom-color: #000;
    }

    &:hover:after {
      cursor: pointer;
      border-top-color: #000;
    }
  }
}
