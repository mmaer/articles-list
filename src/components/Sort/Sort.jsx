import React from 'react';

import './sort.scss';

const Sort = () => {
  return (
    <div className="sort">
      Sort by date: <span className="sort__arrows" />
    </div>
  );
};

export default Sort;
