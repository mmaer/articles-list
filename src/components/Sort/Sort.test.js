import React from 'react';
import { shallow, mount } from 'enzyme';

import Sort from './';

describe('Sort component', () => {
  it('Should match snapshot', () => {
    const wrapper = shallow(<Sort setOrderByASC={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call setOrderByASC when arrows are clicked', () => {
    const mockSetOrderByASC = jest.fn();
    const wrapper = mount(<Sort setOrderByASC={mockSetOrderByASC} />);
    const arrows = wrapper.find('span');
    arrows.simulate('click');

    expect(mockSetOrderByASC.mock.calls.length).toBe(1);
  });
});
