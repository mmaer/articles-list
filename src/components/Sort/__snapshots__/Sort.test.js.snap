// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort component Should match snapshot 1`] = `
<div
  className="sort"
>
  Sort by date:
  <span
    className="sort__arrows"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="0"
  />
</div>
`;
